### DISCUSSION TITLE
examples of integrating 'mem' regions and virtual registers?

### DISCUSSION THREAD

**Discussion by ekallal** [SystemRDL/systemrdl-compiler#208] [Q&A] [answered]
Does anybody have examples of integrating a mem region and virtual registers into an addrmap?  From [this issue](https://github.com/SystemRDL/PeakRDL-regblock/issues/36)  and the systemRDL spec, it seems like this could be powerful.  I haven't been able to figure out how virtual registers impact the generated RTL. 

This search was inspired by the latest regblock release which added fields in the hwif struct for external registers - thought there might be something similar with mem.  
https://github.com/SystemRDL/PeakRDL-regblock/issues/84

Thanks!

**Comment by amykyta3** [comment, marked as answer]
`mem` components are always interpreted as external to the register file. You can still generate a design that has a `mem` instance in it, but the regblock generator will only go as far as create a hardware interface that you use to connect to the memory structure. It is up to you to instantiate the technology-specific memory and complete the connections.

See the docs on how external components work: https://peakrdl-regblock.readthedocs.io/en/latest/rdl_features/external.html
